src/main/java/org/broadinstitute/hellbender/engine/spark/GATKRegistrator.java

```diff
@@ -43,6 +43,13 @@ public void registerClasses(Kryo kryo) {
         kryo.register(SAMRecordToGATKReadAdapter.class, new SAMRecordToGATKReadAdapterSerializer());
 
         kryo.register(SAMRecord.class, new SAMRecordSerializer());
+        kryo.register(BAMRecord.class, new SAMRecordSerializer());
+
+        kryo.register(SAMFileHeader.class);
```

Comment:
This collides (admittedly harmlessly, in this case) with a registration for the same class in the ADAM serializer. I think we have to deal with the issue of registration precedence in this branch -- can you find a way to ensure that the GATK registrations always take precedence over any registrations for the same classes in our dependencies?